tree/dataframe/test/dataframe_vary.cxx

```diff
@@ -662,84 +862,337 @@ TEST(RDFVary, VaryDisplay) // TEST instead of TEST_P because Display is single-t
+   }
+};
 
-}
+TEST_P(RDFVary, VaryCustomObject)
+{
+   // this object can be filled
+   auto h = ROOT::RDataFrame(10)
+                   .Define("Jet",
+                           [] {
+                              return Jet{1., 2.};
+                           })
+                   .Vary(
+                      "Jet",
+                      [] {
+                         return ROOT::RVec<ROOT::RVecD>{{}, {4., 5.}};
+                      },
+                      {}, 2)
+                   .Fill<Jet>(CustomFiller{}, {"Jet"});
 
-TEST_P(RDFVary, DISABLED_VaryHisto2D)
-{
+   auto hs = VariationsFor(h);
 
-}
+   EXPECT_DOUBLE_EQ(h->GetEntries(), 10.);
+   EXPECT_DOUBLE_EQ(h->GetMeanX(), 1.);
+   EXPECT_DOUBLE_EQ(h->GetMeanY(), 2.);
 
-TEST_P(RDFVary, DISABLED_VaryHisto3D)
-{
+   EXPECT_DOUBLE_EQ(hs["nominal"].GetEntries(), 10.);
+   EXPECT_DOUBLE_EQ(hs["nominal"].GetMeanX(), 1.);
+   EXPECT_DOUBLE_EQ(hs["nominal"].GetMeanY(), 2.);
+
+   EXPECT_DOUBLE_EQ(hs["x:0"].GetEntries(), 10.);
+   EXPECT_DOUBLE_EQ(hs["x:0"].GetMeanX(), 0.);
+   EXPECT_DOUBLE_EQ(hs["x:0"].GetMeanY(), 0.);
 
+   EXPECT_DOUBLE_EQ(hs["x:1"].GetEntries(), 10.);
+   EXPECT_DOUBLE_EQ(hs["x:1"].GetMeanX(), 4.);
+   EXPECT_DOUBLE_EQ(hs["x:1"].GetMeanY(), 5.);
 }
 
-TEST_P(RDFVary, DISABLED_VaryHistoND)
+TEST_P(RDFVary, VaryFill)
 {
-
+   SimpleFiller sf;
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [] { return 42.; })
+               .Vary(
+                  "x",
+                  [](double x) {
+                     return ROOT::RVecD{x - 1., x + 1.};
+                  },
+                  {"x"}, 2)
+               .Fill<double>(sf, {"x"});
+   EXPECT_THROW(VariationsFor(h), std::logic_error); // FIXME not (yet) implemented
 }
 
-TEST_P(RDFVary, DISABLED_VaryMax)
+TEST_P(RDFVary, VaryGraph)
 {
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Graph<int, int>("x", "x");
+   auto hs = VariationsFor(h);
 
+   EXPECT_DOUBLE_EQ(h->GetMean(), 4.5);
+   EXPECT_DOUBLE_EQ(hs["nominal"].GetMean(), 4.5);
+   EXPECT_DOUBLE_EQ(hs["x:0"].GetMean(), 3.5);
+   EXPECT_DOUBLE_EQ(hs["x:1"].GetMean(), 5.5);
 }
 
-TEST_P(RDFVary, DISABLED_VaryMean)
+TEST_P(RDFVary, VaryGraphAsymmErrors)
 {
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .GraphAsymmErrors<int, int, int, int, int, int>("x", "x", "x", "x", "x", "x");
+   EXPECT_DOUBLE_EQ(h->GetMean(), 4.5);
+   EXPECT_THROW(VariationsFor(h), std::logic_error); // FIXME not (yet) implemented
```

Comment:
same here, please write the full test, then we can `DISABLE` it with a pointer to the github issue